pkg/kv/kvserver/replica.go

```diff
@@ -1102,7 +1102,7 @@ func (r *Replica) checkExecutionCanProceed(
 		return err
 	} else if err := r.checkSpanInRangeRLocked(ctx, rSpan); err != nil {
 		return err
-	} else if err := r.checkTSAboveGCThresholdRLocked(ba.Timestamp, st, ba.IsAdmin()); err != nil {
+	} else if err := r.checkTSAboveGCThresholdRLocked(ba.EarliestTimestamp(), st, ba.IsAdmin()); err != nil {
```

Comment:
Not sure when `checkExecutionCanProceedForRangeFeed` is used but do I need to do this there too?